sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala

```diff
@@ -144,19 +143,10 @@ class CacheManager extends Logging {
       } else {
         _.sameResult(plan)
       }
-    val plansToUncache = mutable.Buffer[CachedData]()
-    readLock {
-      val it = cachedData.iterator()
-      while (it.hasNext) {
-        val cd = it.next()
-        if (shouldRemove(cd.plan)) {
-          plansToUncache += cd
-        }
-      }
-    }
+    val plansToUncache = cachedData.filter(cd => shouldRemove(cd.plan))
```

Comment:
nit:
```
val (plansToUncache, remainingPlans) = cachedData.partition(cd => shouldRemove(cd.plan))
...
writeLock {
  cachedData = remainingPlans
}
...
```

Reply:
Ah, much nicer.  Will refactor the other spot too